### ISSUE TITLE
Style/FetchEnvVar - false positive cases with (not) equal operators

### ISSUE THREAD

**Issue by deepj** [rubocop/rubocop#10581]
## Expected behavior

No issue while using ENVs and (not) equal operators

```ruby
if ENV['DISABLE_DOTENV'] != '1'
  # Some Ruby code
end
```

## Actual behavior

```
Style/FetchEnvVar: Use ENV.fetch('DISABLE_DOTENV') or ENV.fetch('DISABLE_DOTENV', nil) instead of ENV['DISABLE_DOTENV'].
if ENV['DISABLE_DOTENV'] != '1'
   ^^^^^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

Code snippet

```ruby
if ENV['DISABLE_DOTENV'] != '1'
  # Some Ruby code
end
```

Run

```
rubocop --only Style/FetchEnvVar
```

## RuboCop version

```
$ rubocop -V
1.28.2 (using Parser 3.1.2.0, rubocop-ast 1.17.0, running on ruby 3.1.2 arm64-darwin21)
  - rubocop-performance 1.13.3
  - rubocop-rails 2.14.2
```
